sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamJoinRel.java

```diff
@@ -369,8 +370,8 @@ private Row buildNullRow(BeamRelNode relNode) {
   private PCollection<Row> joinAsLookup(
       BeamRelNode leftRelNode,
       BeamRelNode rightRelNode,
-      PCollectionTuple inputPCollections) {
-    PCollection<Row> factStream = inputPCollections.apply(leftRelNode.toPTransform());
+      PInput inputPCollections) {
+    PCollection<Row> factStream = inputPCollections.getPipeline().apply(leftRelNode.toPTransform());
```

Comment:
`getPipeline().apply()` is probably not what you want here, either. It is actually a bad method - it is the same as `getPipeline().begin().apply()` so it always starts a new initial pipeline segment.

Reply:
The current model is wrong, the new model is wrong. I've dropped this change and just added another `PCollectionTuple.empty(input.getPipeline())`. We can discuss the right way to do it as a followup.